libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/marker/RoutePreviewLongPressMapComponent.kt

```diff
@@ -36,12 +38,14 @@ internal class RoutePreviewLongPressMapComponent(
     }
 
     private val longClickListener = OnMapLongClickListener { point ->
-        val location = store.state.value.location?.enhancedLocation
-        location?.toPoint()?.also { lastPoint ->
-            store.dispatch(DestinationAction.SetDestination(Destination(point)))
-            store.dispatch(RoutePreviewAction.FetchPoints(listOf(lastPoint, point)))
-            hapticFeedback?.tick()
-        } ?: logW(TAG, "Current location is unknown so map long press does nothing")
+        if (context.options.enableMapLongClickIntercept.value) {
```

Comment:
I don't like that there are multiple places where this flag is checked. It's easy to forget in the future about it. As for now, it is checked in 2 listeners. What would you say to sth like this?
We don't check it in listeners, instead, we check it in `onAttached` and depending on the value of this flag we either register our listener or not. In order not to check it in 2 places, we introduce a new abstract class: `LongClickableUIComponent` (draft name) that extends `UIComponent` and has `mapView` and `abstract longClickListener` properties. While `FreeDriveLongPressMapComponent` and `RoutePreviewLongPressMapComponent` extend `LongClickableUIComponent`. Thus we get only one place where the flag changes the behaviour. WDYT?